solutions/swb-reference/integration-tests/tests/isolated/sshKeys/sendSshKey.test.ts

```diff
@@ -0,0 +1,189 @@
+  const validLaunchParameters = {
+    name: randomTextGenerator.getFakeText('name'),
+    description: randomTextGenerator.getFakeText('send ssh key test description'),
+    envTypeId: setup.getSettings().get('envTypeId'),
+    envTypeConfigId: setup.getSettings().get('envTypeConfigId'),
+    datasetIds: [],
+    envType: setup.getSettings().get('envType')
+  };
+
+  const testBundle = [
+    {
+      username: 'projectAdmin1',
+      session: () => pa1Session,
+      projectId: () => project1Id
+    },
+    {
+      username: 'researcher1',
+      session: () => rs1Session,
+      projectId: () => project1Id
+    }
+  ];
+
+  beforeAll(async () => {
+    ({ adminSession, pa1Session, rs1Session, pa2Session, project1Id, project2Id, anonymousSession } =
+      await paabHelper.createResources(__filename));
+    const { data: env1 } = await pa1Session.resources.projects
+      .project(project1Id)
+      .environments()
+      .create(validLaunchParameters, false);
+    project1EnvId = env1.id;
+    const { data: env2 } = await pa2Session.resources.projects
+      .project(project2Id)
+      .environments()
+      .create(validLaunchParameters, false);
+    project2EnvId = env2.id;
+  });
+
+  beforeEach(async () => {
+    expect.hasAssertions();
+  });
+
+  afterAll(async () => {
+    await paabHelper.cleanup();
+  });
+
+  describe(`when project id does not match env's project`, () => {
+    // project is project1 and env is project2 but each user has access to project1
+    test.each(testBundle)('it throws 400', async (testCase) => {
+      const { username, session: sessionFunc, projectId: projectIdFunc } = testCase;
+      const session = sessionFunc();
+      const projectId = projectIdFunc();
+
+      console.log(`as ${username}`);
+
+      try {
+        await session.resources.projects
+          .project(projectId)
+          .environments()
+          .environment(project2EnvId)
+          .sshKeys();
+      } catch (e) {
+        checkHttpError(
+          e,
+          new HttpError(400, {
+            error: 'Bad Request',
+            message: `Requested Project ID ${projectId} does not match environment Project ID ${project2Id}`
+          })
+        );
+      }
+    });
+  });
+
+  describe(`when instance id is not defined`, () => {
+    test.each(testBundle)('it throws 400', async (testCase) => {
+      const { username, session: sessionFunc, projectId: projectIdFunc } = testCase;
+      const session = sessionFunc();
+      const projectId = projectIdFunc();
+
+      console.log(`as ${username}`);
+
+      try {
+        await session.resources.projects
+          .project(projectId)
+          .environments()
+          .environment(project1EnvId)
+          .sshKeys();
+      } catch (e) {
+        checkHttpError(
+          e,
+          new HttpError(400, {
+            error: 'Bad Request',
+            message: `Instance Id is not defined for environment ${project1EnvId} yet. Try again later.`
```

Comment:
resourceId are all removed from error message. refer to https://github.com/aws-solutions/research-service-workbench-on-aws/pull/1146

Reply:
sick catch! will update!